genomepy/functions.py

```diff
@@ -154,7 +155,9 @@ def install_genome(
     genome_dir=None,
     localname=None,
     mask="soft",
+    toplevel=False,
```

Comment:
toplevel is probably specific top Ensembl (maybe also NCBI)? In that case we should not have it as an argument to `install_genome()` as it is provider-specific. Maybe use a `**kwargs` construct?

Reply:
Thinking about this some more.. This warrants a general solution. What to do with provider-specific options. There is an additional option for Ensembl, `version`, which is currently not exposed in the command line tool. We need a general solution that ideally passes all provider-specific option through to the provider-specific functions.


For instance, at the command line:

```
genomepy install genome ensembl --ensembl-version 92 --ensembl-toplevel
```

Then all arguments that start with the provider name would be passed to that provider. The provider should provide these extra options, which could then be retrieved in  `genomepy/cli.py`.

Reply:
The human genome on NCBI is stored in top-level format indeed ([source](ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13/README.txt)). Not sure if there is a primary assembly however. 

I will look into this next week.

Reply:
I don't think this solution is supported _properly_ by `click`. Here's what I found:

Ideally we'd get a list of provider-specific options, and then iteratively add `@click.option()`s for each, thus providing a clear and consistent help menu. However, calling `@click` functions in a loop does not seem to work as it required all lines after `@click.command()` to start with either `@` or `def`.

One `@click.argument("provider-specific options", nargs=-1)` can accept any number of arguments, but the options wouldn't be documented. The explanation would have to be in the docstrings. The docstrings cannot have single newlines, so all options would either be in sequence or spaced with a whiteline between each. And the docstrings cannot be extended with functions.

The only remaining solution would be to add one `@click.option()` that accepts a string input. The help text of options can be extended using `.format()`, but now the options and their explanations would be squeezed on half of the screen.

Reply:
What about the options detailed here:

- https://stackoverflow.com/questions/50499340/specify-options-and-arguments-dynamically
- https://stackoverflow.com/questions/53147525/is-it-possible-to-dynamically-generate-commands-in-python-click

I think especially the first one would work. You can create a custom decorator for the `genomepy install` command.

Reply:
Excellent suggestion. It worked! 

Current solution has a dictionary with all options in `cli.py`. This dictionary contains provider-specific options as well. 

About the idea to let each provider provide the options: I'm not sure how much cleaner this is. 

Right now, all options are in `cli.py`. When a new option is added, these can be added in `cli-py`, and extracted from `kwargs` in `providers.py`. No changes from the current workflow. 

However, if these were placed elsewhere (either in `providers.py` or even in a different location in `cli.py`), then the location for these options depends on whether they are general or specific. More messy in my opinion.

Reply:
Opinion overruled :p 

Added provider specific options to `providers.py`!